test/integration/builder/inserts.js

```diff
@@ -1465,6 +1465,110 @@ module.exports = function (knex) {
       expect(rows[0].name).to.equal('AFTER');
     });
 
+    it('conditionally updates rows when inserting a duplicate key to unique column and merge with where clause matching row(s) is specified', async function () {
+      if (/redshift/i.test(knex.client.driverName)) {
+        return this.skip();
+      }
+
+      // Setup: Create table with unique email column
+      await knex.schema.dropTableIfExists('upsert_tests');
+      await knex.schema.createTable('upsert_tests', (table) => {
+        table.string('name');
+        table.string('email');
+        table.string('role');
+        table.unique('email');
+      });
+
+      // Setup: Create row to conflict against
+      await knex('upsert_tests').insert({
+        email: 'mergetest@example.com',
+        role: 'tester',
+        name: 'BEFORE',
+      });
+
+      // Perform insert..merge (upsert)
+      try {
+        await knex('upsert_tests')
+          .insert({ email: 'mergetest@example.com', name: 'AFTER' }, 'email')
+          .onConflict('email')
+          .merge()
+          .where('upsert_tests.role', 'tester')
+          .testSql(function (tester) {
+            tester(
+              'pg',
+              'insert into "upsert_tests" ("email", "name") values (?, ?) on conflict ("email") do update set "email" = excluded."email", "name" = excluded."name" where "upsert_tests"."role" = ? returning "email"',
+              ['mergetest@example.com', 'AFTER', 'tester']
+            );
+          });
+      } catch (err) {
+        if (/oracle|mssql/i.test(knex.client.driverName)) {
```

Comment:
Skipping test altogether same as is done for `redshift` probably would be more transparent in test results log.

Reply:
I don't fully understand the test setup - but it seems like the difference between redshift and oracle/mssql is that for oracle/mssql we expect an error to be thrown, but not redshift? Should I preserve that distinction or just add mssql/oracle to the skip?

Reply:
TBH, I think we don't even run redshift in CI, so feel free to just keep ignoring it.
BTW, shouldn't MySQL also be included in error assertion?

Reply:
Ah yeah. Added MySQL to have an error assertion:

90cba493009f09ceb4457c4040cfb1da406129d2